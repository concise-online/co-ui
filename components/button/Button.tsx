import React, { ButtonHTMLAttributes } from 'react';
import classNames from 'classnames';
import { getPrefixCls } from '../utils/config';

const prefixCls = getPrefixCls('btn');

interface Props extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type'> {
  type?: 'primary' | 'success' | 'info' | 'warning' | 'danger';
  htmlType?: 'button' | 'submit' | 'reset';
  fill?: boolean;
}

function Button(props: Props) {
  const { children, className, htmlType, type, fill, ...btnProps } = props;
  const classes = classNames(prefixCls, className, {
    [`${prefixCls}-${type}`]: !!type,
    [`${prefixCls}-fill`]: fill,
  });

  return (
    <button
      type={htmlType}
      className={classes}
      {...btnProps}
    >
      <span>{children}</span>
    </button>
  );
}

export default Button;
