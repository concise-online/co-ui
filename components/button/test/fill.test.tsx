import React from 'react';
import { render } from 'enzyme';
import Button from '../index';

describe('button fill feature', () => {
  it('renders correctly', () => {
    const wrapper = render(<Button fill>Button</Button>);

    expect(wrapper).toMatchSnapshot();
  });
});
