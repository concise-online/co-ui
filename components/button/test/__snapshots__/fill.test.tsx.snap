// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button fill feature renders correctly 1`] = `
<button
  class="co-btn co-btn-fill"
>
  <span>
    Button
  </span>
</button>
`;
